index.html

```diff
@@ -2296,6 +2296,41 @@ <h1>Core Representations</h1>
 representations should be added here once that section has been written.
     </p>
 
+    <section>
+      <h2>Handling Properties</h2>
+      
+        <section>
+          <h2>Unknown Properties</h2>
+          <p>
+    When a DID Document is created by consuming a compliant representation, any properties
+    in the representation that are not known to the consumer MUST be added to the
+    data model as properties without any further processing on the property
+    value.
+          </p>
+          <p>
+    When a DID Document is serialized by a producer into a compliant
+    representation, any unknown properties MUST be serialized into the
+    representation. When the data type for the property is unknown by the producer,
+    a string serialization of the property SHOULD be used.
```

Comment:
how is it possible that the abstract data model would have properties of unknown types?... seems like a security issue to serialize an unknown type to a string.... `Buffer.from("40636f6e74656e7874", "hex").toString()`...

Reply:
Same question -- if an unknown property value can't be converted into the abstract data model such that it can be re-exported to the same representation losslessly, there is a fundamental problem. You can either take the input representation and losslessly place it into the abstract data model (so you can at least pump it back out to the original incoming representation) or you can't. If you can't, you should probably throw an error.

Reply:
The abstract data model is going to have a type for every property, because it's in the abstract data model. 

This is why I have argued for the properties within the data model to have abstract types defined, and all representations have clear and deterministic mappings for all known types into and out of the serializations.

Reply:
> The abstract data model is going to have a type for every property, because it's in the abstract data model.

This PR states:

> When a DID Document is created by consuming a compliant representation, any properties in the representation that are not known to the consumer MUST be added to the data model as properties without any further processing on the property value.

It won't have a type for unknown properties... right? What am I missing?

Reply:
I would have loved for formal types for the abstract data model... I even tried to help define them using JSON Schema... nobody contributed to it... I removed it.

If the abstract data model cannot represent a type, it cannot support representations that rely on that type... which means right now, that we have an abstract data model with no type system, and no way to actually answer this question....

If we say the abstract data model is modeling JSON, we know it supports JSON types... if we say its INFRA, we know it supports infra types... 

It's not possible for it to have a property with an unknown type.... unless the abstract data model itself does not live in a type system....

What type system does the abstract data model live in? Haskell? INFRA? JSON? CBOR?... if we don't know, we can't answer the question about which representations are possible with the abstract data model.

Haskell would work for everything (supports CBOR / JSON)...

JSON would not work for CBOR.

CBOR would work for JSON.

INFRA.... TBD... not sure about .... numbers... lol.

Imo this is solved in 1 of 2 ways...

either we define the abstract data model as anything that can be represented in JSON (for now, and later we fix that to allow CBOR / ProtoBuff)....

or we forbid "unknown" properties from being round tripped from it... because serializing an unknown type to a string is not only unstable, its dangerous.... what happens to protobuff / cbor ?... a bunch of nonsense binary encoded as a utf-8 string?...doesn't that mean we can't round trip a binary format?

I would actually be in favor of saying that JSON is the abtract data model (for now)... then this problem goes away for JSON and JSON-LD.

In the future, we can define a new abstract data model that is a super set of JSON that can be mapped to it... and have backwards compatibility :)

Reply:
The abstract data model currently uses Infra types for all declared properties, as well as metadata, so I don't see why we wouldn't do that here. We also have additional formats like URIs and Dates that are easily encoded as formatted strings. 

When you create the abstract data model, your source for that (JSON, CBOR, Whatever) will have a data type associated with it. The consumption rules MUST declare a default abstract data type to map that concrete data type to. Same idea with the production direction: all representations declare rules for mapping all allowed abstract data types.

Reply:
so I assume infra is capable of preserving strings, arrays and maps.... so let it preserve them, and let did method authors decide if they want to include processing directives or not in THEIR json :) its not for this WG to redefine what JSON is.